tests/BlazingSQLTest/EndToEndTests/crossJoinsTest.py

```diff
@@ -0,0 +1,147 @@
+                              use_percentage, fileSchemaType)
+
+            queryId = 'TEST_02'
+            query = """
+                    select o_orderkey, o_totalprice,
+                         l_linenumber, l_shipmode
+                    from orders cross join lineitem
+                    where o_orderkey < 6
+                    and l_receiptdate > date '1996-07-12'
+                    and l_linenumber > 5
+                    and o_totalprice < 74029.55
+                    and o_clerk = 'Clerk#000000880'
+                    and l_shipmode IN ('FOB', 'RAIL')
+                    order by o_orderkey, o_totalprice, l_linenumber"""
+            runTest.run_query(bc, spark, query, queryId, queryType,
+                              worder, '', acceptable_difference,
+                              use_percentage, fileSchemaType)
+
+            queryId = 'TEST_03'
+            query = """select o_orderkey, n_nationkey
+                    from nation cross join orders
+                    where o_totalprice > 4000.0
+                    and o_orderdate > date '1998-07-12'
+                    and o_orderkey > 425000
+                    group by o_orderkey, n_nationkey
+                    order by o_orderkey, n_nationkey"""
+            runTest.run_query(bc, spark, query, queryId, queryType, worder,
+                              '', acceptable_difference,
+                              use_percentage, fileSchemaType)
+
+            queryId = 'TEST_04'
+            query = """
+                    with cust_nation as
+                    (
+                        select c_custkey, c_name, n_nationkey, n_name
+                        from customer inner join nation
+                        on c_nationkey = n_nationkey
+                        where n_nationkey > 21
+                        and c_acctbal > 525.0
+                        and c_custkey > 13450
+                        order by c_custkey, n_nationkey
+                    ), ord_lineitem as
+                    (
+                        select o_orderkey, l_quantity
+                        from orders left join lineitem
+                        on o_orderkey = l_orderkey
+                        where l_shipdate > date '1998-11-12'
+                        and o_totalprice > 3500.0
+                        and l_quantity > 48.0
+                        and l_shipmode in ('AIR', 'FOB', 'SHIP')
+                        order by o_orderkey
+                    )
+                    select c_custkey, n_name, l_quantity
+                    from ord_lineitem cross join cust_nation
+                    where n_name = 'RUSSIA'
+                    order by c_custkey"""
+            runTest.run_query(bc, spark, query, queryId, queryType, worder,
+                              '', acceptable_difference,
+                              use_percentage, fileSchemaType)
+
+            if Settings.execution_mode == ExecutionMode.GENERATOR:
```

Comment:
There is a query that is part of TPC-H that does not explicitly have a cross join, but the relational algebra created for that query does have one. I think the query has an EXIST clause. We should make sure that query is now part of our end to end tests

Reply:
The query 04 (TPCH) as you reported from #702 . But, this doesn't have a `condition=[true]` statement. The optimized plan is here:
```
LogicalSort(sort0=[$0], dir0=[ASC])
  LogicalAggregate(group=[{0}], order_count=[COUNT()])
    LogicalProject(o_orderpriority=[$1])
      **LogicalJoin(condition=[=($0, $2)], joinType=[inner])**
        LogicalProject(o_orderkey=[$0], o_orderpriority=[$2])
          BindableTableScan(table=[[main, orders]], filters=[[AND(>=($1, 1993-07-01 00:00:00), <($1, 1993-10-01 00:00:00))]], projects=[[0, 4, 5]], aliases=[[o_orderkey, $f1, o_orderpriority]])
        LogicalProject(l_orderkey=[$0])
          LogicalAggregate(group=[{0}], agg#0=[MIN($1)])
            LogicalProject(l_orderkey=[$0], $f0=[true])
              BindableTableScan(table=[[main, lineitem]], filters=[[AND(<($1, $2), IS NOT NULL($0))]], projects=[[0, 11, 12]], aliases=[[l_orderkey, $f1, $f2]]) 
```